src/mpid/common/genq/mpidu_genq_shmem_pool.c

```diff
@@ -31,6 +31,11 @@ static int cell_block_alloc(MPIDU_genqi_shmem_pool_s * pool, int block_idx)
     /* init cell headers */
     int idx = block_idx * pool->cells_per_proc;
     for (int i = 0; i < pool->cells_per_proc; i++) {
+        /* Have the "host" process for each cell zero-out the cell contents to force the first-touch
+         * policy to make the pages resident to that process. */
+        memset((char *) pool->cell_header_base + (idx + i) * pool->cell_alloc_size, 0,
+               pool->cell_alloc_size);
```

Comment:
Do we need additional instruction to pin the memory? Otherwise, the kernel is free to swap out the memory anytime, right? Although how likely is that is a separate investigation.

Reply:
So my understanding after talking to smart people at Intel that work on this sort of thing is that relying on the first touch policy is sufficient. I looked at other options using the POSIX NUMA APIs, but they either didn't accomplish what we were looking for or were much more complex without providing any further guarantees.

Reply:
This discussion should move to https://github.com/pmodels/mpich/pull/5759.